src/app/child-dev-project/notes/notes-manager/notes-manager.component.ts

```diff
@@ -80,8 +83,24 @@ export class NotesManagerComponent implements OnInit {
     });
     this.entityMapperService.loadType<Note>(Note).then((notes) => {
       notes.forEach((note) => (note["color"] = this.getColor(note)));
-      this.notes = notes;
+      // This prevents edge-cases where updates are received
+      // before this is
+      this.notes = notes.concat(this.notes);
     });
+
+    this.entityMapperService
+      .receiveUpdates<Note>(Note)
+      .pipe(
+        untilDestroyed(this),
+        filter((note) => note !== null && note !== undefined),
```

Comment:
This happens whenever the component initially subscribes to the observable and `null` is emitted due to the implementation of `BehaviorSubject`. I don't know if there is a better way (probably) but I couldn't find it.

Reply:
So I googled around a little and found the concept of [hot and cold observables](https://stackoverflow.com/questions/32190445/hot-and-cold-observables-are-there-hot-and-cold-operators). A cold observable means, that every subscriber will get all the emitted values, even the ones that are emitted before they have subscribed. When subscribing to a hot observable, only the values which are emitted after the subscription happened, will trigger the subscriber. For this use case I believe, that subscribers should only be notified about updates that happen _after_ they subscribed (because the previous updates are already in the database), therefore a behavior subject is probably the wrong choice because it always emits one old value. Instead a simple `Observer` (which is a hot observable) should do the trick, that no old values are emitted and hence no null value should ever arrive.

One more thing to note: besides the `Subject`, all RxJS observers are lazy, which means that the first subscriber will get all values that have been emitted since the observable has been created. In this case I don't think, the laziness will be a problem, because the observers are only created when someone wants to subscribe to it.

Reply:
I opted for a `Subject` instead. This makes sure, more than one receiver can receive the values emitted, while not having the "behavioral" part. A simple `Observable` would require to think about the edge-case where more than one entity subscribes to updates of the same entity-type.